ArchitectureFlow/ArchitectureFlow/Login/UseCase/Login.h

```diff
@@ -0,0 +1,11 @@
+#import <Foundation/Foundation.h>
+
+@class Promise;
+@class LoginRequestModel;
+@protocol LoginOutput;
+
+@protocol Login
```

Comment:
Would it be better to explicitly call this LoginUsecase? Seeing this in the code without context could be confusing, especially with less obvious examples.

Reply:
https://confluence.skyscannertools.net/display/AG/1.+Clean+Architecture+Knowledge+Sharing

Reply:
I Recommend rewatching my presentations as a lot of questions here I've talked through extensively there. 
1) LoginUseCase name convention: We could of course call it LoginUseCase. This is an Input Boundary (please prewatch the presentation). As you know the use case comes in a form of a command design pattern, ergo it's name is a command sentence, like Login, Logout, SendMessage. I think it's easy enough to find them, because one can just look for the word Interactor. Also - if we had a good package structure, they will be all on the domain layer. But it's not a big deal to call it UseCase either
2) Does Interactor implement usecase so 1 interactor implements only 1 usecase, but the usecase can be implemented by multiple interactors (so the other way around). The protocol is the InputBoundary protocol from clean architecture and the sole reason for it's existence is to provide an abstraction so that nothing points inwards into the actual implementation, the LoginInteractor. When you look at the clean diagram: 
![image](https://user-images.githubusercontent.com/23355816/36192817-32d64e52-115b-11e8-9f8f-d8a6af84b56e.png) notice how not a single arrow points into the interactor. That means that the caller depends on abstraction, the InputBoundary to be exact. 
3) Is interactor stateless: Again, I recommend rewatching my presentation. It's imporatant interactors as stateless (https://groups.google.com/forum/#!topic/clean-code-discussion/S9iHnnpGXAA). This is because Ineractor is not master of the data. It performs operations on data or entities, but it doesn't own them. Same interactor can be executed multiple times and each time should be intependent from one another from the interactor point of view. Repositories however are the masters of data and state is stored there typically. This is also how we achieve offline support

Since it's a no-op, we can skip the `Login` protocol alltogether and only have LoginInteractor. Then it's both easy to find and there is 1 less protocol to define. That would mean that one of the dependency arrows would point directly at the interactor, but it will not impact any functionality or testability so we could do that